File: postman/sdks/Collections SDKs/tagslut - API/typescript/src/services/validation-run/validation-run-service.ts
import { z } from 'zod';
import { BaseService } from '../base-service';
import { ContentType, HttpResponse, SdkConfig } from '../../http/types';
import { RequestBuilder } from '../../http/transport/request-builder';
import { SerializationStyle } from '../../http/serialization/base-serializer';
import { ThrowableError } from '../../http/errors/throwable-error';
import { Environment } from '../../http/environment';
import { _6aResolveTidalAlbumToIsrcParams } from './request-params';

/**
 * Service class for ValidationRunService operations.
 * Provides methods to interact with ValidationRunService-related API endpoints.
 * All methods return promises and handle request/response serialization automatically.
 */
export class ValidationRunService extends BaseService {
  protected _6aResolveTidalAlbumToIsrcConfig: Partial<SdkConfig> = { environment: Environment.API };

  protected _6bTrackByIdValidation_Config: Partial<SdkConfig> = {
    environment: Environment.BASE_URL,
  };

  protected _6cRunNotesConfig: Partial<SdkConfig> = { environment: Environment.EXAMPLE };

  /**
   * Sets method-level configuration for _6aResolveTidalAlbumToIsrc.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_6aResolveTidalAlbumToIsrcConfig(config: Partial<SdkConfig>): this {
    this._6aResolveTidalAlbumToIsrcConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for _6bTrackByIdValidation_.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_6bTrackByIdValidation_Config(config: Partial<SdkConfig>): this {
    this._6bTrackByIdValidation_Config = config;
    return this;
  }

  /**
   * Sets method-level configuration for _6cRunNotes.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_6cRunNotesConfig(config: Partial<SdkConfig>): this {
    this._6cRunNotesConfig = config;
    return this;
  }

  /**
   *
   * @param {string} [params.countryCode] -
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _6aResolveTidalAlbumToIsrc(
    params?: _6aResolveTidalAlbumToIsrcParams,
    requestConfig?: Partial<SdkConfig>,
  ): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(
      this._6aResolveTidalAlbumToIsrcConfig,
      requestConfig,
    );
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/v1/albums/507881809/tracks')
      .setRequestSchema(z.any())
      .addAccessTokenAuth(resolvedConfig?.token, 'Bearer')
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addQueryParam({
        key: 'countryCode',
        value: params?.countryCode,
      })
      .build();
    return this.client.callDirect<any>(request);
  }

  /**
   *
   * @param {string} beatportTestTrackId -
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _6bTrackByIdValidation_(
    beatportTestTrackId: string,
    requestConfig?: Partial<SdkConfig>,
  ): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(
      this._6bTrackByIdValidation_Config,
      requestConfig,
    );
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/v4/catalog/tracks/{beatport_test_track_id}')
      .setRequestSchema(z.any())
      .addAccessTokenAuth(resolvedConfig?.token, 'Bearer')
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addPathParam({
        key: 'beatport_test_track_id',
        value: beatportTestTrackId,
      })
      .build();
    return this.client.callDirect<any>(request);
  }

  /**
   *
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _6cRunNotes(requestConfig?: Partial<SdkConfig>): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(this._6cRunNotesConfig, requestConfig);
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/')
      .setRequestSchema(z.any())
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .build();
    return this.client.callDirect<any>(request);
  }
}

File: archive/postman-march-2026/postman/sdks/Collections SDKs/tagslut - API/java/src/main/java/com/tagslutapisdk/services/ValidationRunService.java
package com.tagslutapisdk.services;

import com.fasterxml.jackson.core.type.TypeReference;
import com.tagslutapisdk.config.RequestConfig;
import com.tagslutapisdk.config.TagslutApiSdkConfig;
import com.tagslutapisdk.exceptions.ApiError;
import com.tagslutapisdk.http.Environment;
import com.tagslutapisdk.http.HttpMethod;
import com.tagslutapisdk.http.ModelConverter;
import com.tagslutapisdk.http.util.RequestBuilder;
import com.tagslutapisdk.models._6aResolveTidalAlbumToIsrcParameters;
import java.util.Optional;
import java.util.concurrent.CompletableFuture;
import lombok.NonNull;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

/**
 * ValidationRunService Service
 */
public class ValidationRunService extends BaseService {

  private RequestConfig _6aResolveTidalAlbumToIsrcConfig = RequestConfig.builder()
    .environment(Environment.API)
    .build();
  private RequestConfig _6bTrackByIdValidationConfig = RequestConfig.builder()
    .environment(Environment.BASE_URL)
    .build();
  private RequestConfig _6cRunNotesConfig = RequestConfig.builder()
    .environment(Environment.EXAMPLE)
    .build();

  /**
   * Constructs a new instance of ValidationRunService.
   *
   * @param httpClient The HTTP client to use for requests
   * @param config The SDK configuration
   */
  public ValidationRunService(@NonNull OkHttpClient httpClient, TagslutApiSdkConfig config) {
    super(httpClient, config);
  }

  /**
   * Sets method-level configuration for {@code _6aResolveTidalAlbumToIsrc}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public ValidationRunService set_6aResolveTidalAlbumToIsrcConfig(RequestConfig config) {
    this._6aResolveTidalAlbumToIsrcConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for {@code _6bTrackByIdValidation}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public ValidationRunService set_6bTrackByIdValidationConfig(RequestConfig config) {
    this._6bTrackByIdValidationConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for {@code _6cRunNotes}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public ValidationRunService set_6cRunNotesConfig(RequestConfig config) {
    this._6cRunNotesConfig = config;
    return this;
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @return response of {@code Object}
   */
  public Object _6aResolveTidalAlbumToIsrc() throws ApiError {
    return this._6aResolveTidalAlbumToIsrc(_6aResolveTidalAlbumToIsrcParameters.builder().build());
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @param requestParameters {@link _6aResolveTidalAlbumToIsrcParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _6aResolveTidalAlbumToIsrc(
    @NonNull _6aResolveTidalAlbumToIsrcParameters requestParameters
  ) throws ApiError {
    return this._6aResolveTidalAlbumToIsrc(requestParameters, null);
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @param requestParameters {@link _6aResolveTidalAlbumToIsrcParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _6aResolveTidalAlbumToIsrc(
    @NonNull _6aResolveTidalAlbumToIsrcParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._6aResolveTidalAlbumToIsrcConfig, requestConfig);
    Request request =
      this.build_6aResolveTidalAlbumToIsrcRequest(requestParameters, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6aResolveTidalAlbumToIsrcAsync() throws ApiError {
    return this._6aResolveTidalAlbumToIsrcAsync(
        _6aResolveTidalAlbumToIsrcParameters.builder().build()
      );
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @param requestParameters {@link _6aResolveTidalAlbumToIsrcParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6aResolveTidalAlbumToIsrcAsync(
    @NonNull _6aResolveTidalAlbumToIsrcParameters requestParameters
  ) throws ApiError {
    return this._6aResolveTidalAlbumToIsrcAsync(requestParameters, null);
  }

  /**
   * Method _6aResolveTidalAlbumToIsrc
   * GET /v1/albums/507881809/tracks
   *
   * @param requestParameters {@link _6aResolveTidalAlbumToIsrcParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6aResolveTidalAlbumToIsrcAsync(
    @NonNull _6aResolveTidalAlbumToIsrcParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._6aResolveTidalAlbumToIsrcConfig, requestConfig);
    Request request =
      this.build_6aResolveTidalAlbumToIsrcRequest(requestParameters, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_6aResolveTidalAlbumToIsrcRequest(
    @NonNull _6aResolveTidalAlbumToIsrcParameters requestParameters,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.API),
      "v1/albums/507881809/tracks"
    )
      .setAccessTokenAuth(resolveAccessToken(resolvedConfig))
      .setOptionalQueryParameter("countryCode", requestParameters.getCountryCode())
      .build();
  }

  /**
   * Method _6bTrackByIdValidation
   * GET /v4/catalog/tracks/{beatport_test_track_id}
   *
   * @param beatportTestTrackId String
   * @return response of {@code Object}
   */
  public Object _6bTrackByIdValidation(@NonNull String beatportTestTrackId) throws ApiError {
    return this._6bTrackByIdValidation(beatportTestTrackId, null);
  }

  /**
   * Method _6bTrackByIdValidation
   * GET /v4/catalog/tracks/{beatport_test_track_id}
   *
   * @param beatportTestTrackId String
   * @return response of {@code Object}
   */
  public Object _6bTrackByIdValidation(
    @NonNull String beatportTestTrackId,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._6bTrackByIdValidationConfig, requestConfig);
    Request request = this.build_6bTrackByIdValidationRequest(beatportTestTrackId, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _6bTrackByIdValidation
   * GET /v4/catalog/tracks/{beatport_test_track_id}
   *
   * @param beatportTestTrackId String
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6bTrackByIdValidationAsync(@NonNull String beatportTestTrackId)
    throws ApiError {
    return this._6bTrackByIdValidationAsync(beatportTestTrackId, null);
  }

  /**
   * Method _6bTrackByIdValidation
   * GET /v4/catalog/tracks/{beatport_test_track_id}
   *
   * @param beatportTestTrackId String
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6bTrackByIdValidationAsync(
    @NonNull String beatportTestTrackId,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._6bTrackByIdValidationConfig, requestConfig);
    Request request = this.build_6bTrackByIdValidationRequest(beatportTestTrackId, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_6bTrackByIdValidationRequest(
    @NonNull String beatportTestTrackId,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.BASE_URL),
      "v4/catalog/tracks/{beatport_test_track_id}"
    )
      .setAccessTokenAuth(resolveAccessToken(resolvedConfig))
      .setPathParameter("beatport_test_track_id", beatportTestTrackId)
      .build();
  }

  /**
   * Method _6cRunNotes
   * GET /
   *
   * @return response of {@code Object}
   */
  public Object _6cRunNotes() throws ApiError {
    return this._6cRunNotes(null);
  }

  /**
   * Method _6cRunNotes
   * GET /
   *
   * @return response of {@code Object}
   */
  public Object _6cRunNotes(RequestConfig requestConfig) throws ApiError {
    RequestConfig resolvedConfig = this.getResolvedConfig(this._6cRunNotesConfig, requestConfig);
    Request request = this.build_6cRunNotesRequest(resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _6cRunNotes
   * GET /
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6cRunNotesAsync() throws ApiError {
    return this._6cRunNotesAsync(null);
  }

  /**
   * Method _6cRunNotes
   * GET /
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _6cRunNotesAsync(RequestConfig requestConfig) throws ApiError {
    RequestConfig resolvedConfig = this.getResolvedConfig(this._6cRunNotesConfig, requestConfig);
    Request request = this.build_6cRunNotesRequest(resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_6cRunNotesRequest(RequestConfig resolvedConfig) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.EXAMPLE),
      ""
    ).build();
  }
}
